craft_grammar/models.py

```diff
@@ -18,9 +18,11 @@
 
 import abc
 import re
-from typing import Any, Dict, List, Union
+from typing import Any, Generic, List, TypeVar, get_args, get_origin
```

Comment:
Also don't forget to update the year to 2022-2024 in the copyright notice above.